src/arch/arm/config.cmake

```diff
@@ -260,5 +260,17 @@ add_sources(
 )
 
 add_bf_source_old("KernelArchARM" "structures.bf" "include/arch/arm" "arch/object")
+add_bf_source_old(
+    "KernelArchARM"
```

Comment:
Shouldn't this be "KernelArmHypervisorSupport;KernelArchARM"?

For GICv3 it perhaps makes more sense to test for KernelArmGicV3 directly? And if so, perhaps test for KernelArmGicV2 here as well then?

Reply:
I think the initial patch was more explicit around using KernelArmGicV3, but the refactor patches from @yyshen and @kent-mcleod preferred it this way. I think there is probably another critical look at what is 'GIC' and what is 'GIC extension for virtualization' and where that code belongs. But I really wanted to get the simplest thing that people would agree on in the first instance.

Reply:
This guard is whether a header gets generated for the file.  By not guarding for KernelArmHypervisorSupport the header will always get generated, but it only ever gets included during hypervisor support.  We can make them consistent, but it didn't really reach my threshold for importance when merging patches together.